deploy_snapshot.sh

```diff
@@ -21,28 +21,80 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-if [ "$TRAVIS_REPO_SLUG" != "OfficeDev/ews-java-api" ]; then 
-	echo "[DEPLOY] Skipping snapshot deployment for repo:'$TRAVIS_REPO_SLUG'."
+set -e # Any subsequent(*) commands which fail will cause the shell script to exit immediately
+
+if [ "$TRAVIS_REPO_SLUG" != "OfficeDev/ews-java-api" ]; then
```

Comment:
What is the motivation to block this script for forks?  But default forks won't have travis hooked, and if they do, then we should probably deliver the same behavior.  Independent of whether we die here or not, we should use the environment variable for the rest of the script.